src/libDirectoryService/Coinbase.cpp

```diff
@@ -160,12 +160,14 @@ void DirectoryService::InitCoinbase() {
       ++it;
   }
 
-  const auto& vecLookup = m_mediator.m_lookup->GetLookupNodesStatic();
   const auto& epochNum = m_mediator.m_currentEpochNum;
 
-  for (const auto& lookupNode : vecLookup) {
-    m_coinbaseRewardees[epochNum][CoinbaseReward::LOOKUP_REWARD].push_back(
-        lookupNode.first);
+  if (coinbaseAppendLookup) {
```

Comment:
The reason for this is `InitCoinbase` will be invoked every time a ds trying to `PrepareRunConsensusOnFinalBlockNormal
`, which will also when the failed node received the missing microblocks from leader and restart the consensus. 
If we don't have the flag here, the number of lookup node will duplicate and affect the result of coinbase rewards.

Reply:
How about if we just do `m_coinbaseRewardees[epochNum][CoinbaseReward::LOOKUP_REWARD].clear()` before going through the lookups?

Reply:
good idea